### PR TITLE
integration: bring up tests setup to the latest advancements

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/5278.

### CODE REVIEW

File: integration/setup/setup.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	// Create namespace.
+	{
+		err := config.K8sSetup.EnsureNamespaceCreated(key.Namespace)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if !env.CircleCI() && !env.KeepResources() {
+			defer func() {
```
Comment (kopiczko) [inline integration/setup/setup.go]: Note `defer` is used instead of usual `v == 1` madness.

File: integration/setup/setup.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		err = config.Release.EnsureInstalled(ctx, releaseName, chartInfo, values, installConditions...)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if !env.CircleCI() && !env.KeepResources() {
+			defer func() {
+				deleteConditions := []release.ConditionFunc{
+					config.Release.Condition().CRDNotFound(ctx, releasev1alpha1.NewReleaseCRD()),
+					config.Release.Condition().CRDNotFound(ctx, releasev1alpha1.NewReleaseCycleCRD()),
```
Comment (kopiczko) [inline integration/setup/setup.go]: Here is why I added `CRDNotFound` to the `e2e-harness`.

Comment (kopiczko) [reply integration/setup/setup.go]: I actually need to get rid of that :D CRD is not deleted with the operator.

File: Gopkg.toml [hunk 3]
```diff
@@ -40,7 +40,7 @@ required = [
 
 [[constraint]]
   branch = "master"
-  name = "github.com/giantswarm/e2e-harness"
+  name = "github.com/giantswarm/e2esetup"
```
Comment (xh3b4sd) [inline Gopkg.toml]: NICE

Comment (kopiczko) [reply Gopkg.toml]: Hold your horses :) We still have e2e-harness dependency here. But it's only one const.

Comment (kopiczko) [reply Gopkg.toml]: And whole `pkg/release` package which we are moving in scope of releng SWH...

**Comment by kopiczko** [comment]
This shit works. Finally. Merging.

**Pull request merged**